src/Bedrock.Framework/Transports/Sockets/SocketConnection.cs

```diff
@@ -214,7 +214,10 @@ private async Task<Exception> DoSend()
             finally
             {
                 _aborted = true;
-                _socket.Shutdown(SocketShutdown.Both);
+                if (_socket.Connected)
```

Comment:
This is inherently racy. It makes sense to port the logic from Kestrel back to this client side socket transport.

Reply:
okay then its better you do it since im not familiar with what exactly needs to be ported over from https://github.com/aspnet/AspNetCore/blob/af9f99299cd4a059cf30483d4f7f551ff38c5a28/src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

Reply:
Will do.